crates/ruff_linter/src/rules/flake8_simplify/rules/ast_unary_op.rs

```diff
@@ -25,6 +25,9 @@ use crate::checkers::ast::Checker;
 /// a != b
 /// ```
 ///
+/// ## Fix safety
+/// The fix is marked as unsafe, as `a == b` and/or `a != b` might not return a boolean.
```

Comment:
I think the message is slightly confusing because `a == b` generally returns a boolean, at least for all types that override `__eq__` in a conferment way. 

I'd suggest rephrasing the message to saying that the fix is unsafe for types that override `__eq__` in a non standard way (return something other than `boolean`). I think we did something similar for `round` or some other rule that you worked on